jobrunner/run.py

```diff
@@ -336,21 +343,32 @@ def handle_job(job, api, mode=None, paused=None):
     elif initial_status.state == ExecutorState.FINALIZED:
         # final state - we have finished!
         results = api.get_results(job_definition)
-        save_results(job, job_definition, results)
-        obsolete = get_obsolete_files(job_definition, results.outputs)
-        if obsolete:
-            errors = api.delete_files(job_definition.workspace, Privacy.HIGH, obsolete)
-            if errors:
-                log.error(
-                    f"Failed to delete high privacy files from workspace {job_definition.workspace}: {errors}"
-                )
-            api.delete_files(job_definition.workspace, Privacy.MEDIUM, obsolete)
-            if errors:
-                log.error(
-                    f"Failed to delete medium privacy files from workspace {job_definition.workspace}: {errors}"
+
```

Comment:
Related to comment above. 

How about an early exit:

```
    elif initial_status.state == ExecutorState.FINALIZED:     
        # final state - we have finished!
        if job_definition.cancelled:
            # TODO?: save cancelled job's log (if exists) *somewhere*
            api.cleanup()
            mark_job_as_failed(job, StatusCode.CANCELLED_BY_USER, "Cancelled by user")
            return
         
        results = api.get_results(job_definition)
        save_results(job, job_definition, results)
        ...
   ```
      
This saves a level of indentation and an if branch.

Reply:
yes, that's a nice refactor, done.